vector/src/main/java/im/vector/app/features/home/room/list/home/HomeRoomListFragment.kt

```diff
@@ -0,0 +1,226 @@
+            concatAdapter.addAdapter(getAdapterForData(it))
+        }
+    }
+
+    private fun handleQuickActions(quickAction: RoomListQuickActionsSharedAction) {
+        when (quickAction) {
+            is RoomListQuickActionsSharedAction.NotificationsAllNoisy -> {
+                roomListViewModel.handle(HomeRoomListAction.ChangeRoomNotificationState(quickAction.roomId, RoomNotificationState.ALL_MESSAGES_NOISY))
+            }
+            is RoomListQuickActionsSharedAction.NotificationsAll -> {
+                roomListViewModel.handle(HomeRoomListAction.ChangeRoomNotificationState(quickAction.roomId, RoomNotificationState.ALL_MESSAGES))
+            }
+            is RoomListQuickActionsSharedAction.NotificationsMentionsOnly -> {
+                roomListViewModel.handle(HomeRoomListAction.ChangeRoomNotificationState(quickAction.roomId, RoomNotificationState.MENTIONS_ONLY))
+            }
+            is RoomListQuickActionsSharedAction.NotificationsMute -> {
+                roomListViewModel.handle(HomeRoomListAction.ChangeRoomNotificationState(quickAction.roomId, RoomNotificationState.MUTE))
+            }
+            is RoomListQuickActionsSharedAction.Settings -> {
+                navigator.openRoomProfile(requireActivity(), quickAction.roomId)
+            }
+            is RoomListQuickActionsSharedAction.Favorite -> {
+                roomListViewModel.handle(HomeRoomListAction.ToggleTag(quickAction.roomId, RoomTag.ROOM_TAG_FAVOURITE))
+            }
+            is RoomListQuickActionsSharedAction.LowPriority -> {
+                roomListViewModel.handle(HomeRoomListAction.ToggleTag(quickAction.roomId, RoomTag.ROOM_TAG_LOW_PRIORITY))
+            }
+            is RoomListQuickActionsSharedAction.Leave -> {
+                roomListViewModel.handle(HomeRoomListAction.LeaveRoom(quickAction.roomId))
+                promptLeaveRoom(quickAction.roomId)
+            }
+        }
+    }
+
+    private fun promptLeaveRoom(roomId: String) {
+        val isPublicRoom = roomListViewModel.isPublicRoom(roomId)
+        val message = buildString {
+            append(getString(R.string.room_participants_leave_prompt_msg))
+            if (!isPublicRoom) {
+                append("\n\n")
+                append(getString(R.string.room_participants_leave_private_warning))
+            }
+        }
+        MaterialAlertDialogBuilder(requireContext(), if (isPublicRoom) 0 else R.style.ThemeOverlay_Vector_MaterialAlertDialog_Destructive)
+                .setTitle(R.string.room_participants_leave_prompt_title)
+                .setMessage(message)
+                .setPositiveButton(R.string.action_leave) { _, _ ->
+                    roomListViewModel.handle(HomeRoomListAction.LeaveRoom(roomId))
+                }
+                .setNegativeButton(R.string.action_cancel, null)
+                .show()
+    }
+
+    private fun getAdapterForData(data: HomeRoomSection): EpoxyControllerAdapter {
+        return when (data) {
```

Comment:
Optional: Is there any purpose for this `when` when we only have two cases? A single statement (maybe with `takeIf { it is RoomSummaryData }` might be simpler

Reply:
That's more for a next iterations. There will be more sense in this `when`, when more sections will be added